src/main/java/com/amazon/opendistroforelasticsearch/security/configuration/OpenDistroSecurityIndexSearcherWrapper.java

```diff
@@ -122,11 +122,8 @@ protected final boolean isBlockedIndexRequest() {
     }
 
     protected final boolean isPermittedOnIndex() {
-        final TransportAddress caller = (TransportAddress) this.threadContext.getTransient(ConfigConstants.OPENDISTRO_SECURITY_REMOTE_ADDRESS);
-        final User user = (User) threadContext.getTransient(ConfigConstants.OPENDISTRO_SECURITY_USER);
```

Comment:
Will this automatic object mapping work on some older versions of Java ?

Reply:
Yes, the type inference is available since 1.8 and it is not new for the project, there are few other places where it is already used.